backend/api/user_communities.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"database/sql"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/git-adithyanair/cs130-group-project/token"
+)
+
+func (server *Server) GetUserCommunities(ctx *gin.Context) {
+
+	authPayload := ctx.MustGet(authorizationPayloadKey).(*token.Payload)
```

Comment:
Check that authPayload != nil here before using?

Reply:
authPayload can never be nil, check the middleware.go file